apps/src/lib/node/ledger/shell/finalize_block.rs

```diff
@@ -841,43 +845,71 @@ where
             pgf_inflation_amount.to_string_native()
         );
 
-        // Pgf steward inflation
-        let pgf_stewards_inflation_rate_key =
-            pgf_storage::get_steward_inflation_rate_key();
-        let pgf_stewards_inflation_rate = self
-            .read_storage_key::<Dec>(&pgf_stewards_inflation_rate_key)
-            .unwrap_or_default();
+        let mut pgf_fundings = pgf::get_payments(&self.wl_storage)?;
+        // we want to pay first the oldest fundings
+        pgf_fundings.sort_by(|a, b| a.id.cmp(&b.id));
 
-        let pgf_stewards_pd_rate =
-            pgf_stewards_inflation_rate / Dec::from(epochs_per_year);
-        let pgf_steward_inflation =
-            Dec::from(total_tokens) * pgf_stewards_pd_rate;
+        for funding in pgf_fundings {
+            if credit_tokens(
+                &mut self.wl_storage,
+                &staking_token,
+                &funding.detail.target,
+                funding.detail.amount,
+            )
+            .is_ok()
+            {
+                tracing::info!(
+                    "Minted {} tokens for {} project.",
+                    funding.detail.amount.to_string_native(),
+                    &funding.detail.target,
+                );
+            } else {
+                tracing::info!(
```

Comment:
Should we log this at a different level to make it stand out?

Reply:
good point, maybe `warn` is better